File: packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/nativeProxy/NoopEventHandler.java
package com.swmansion.reanimated.nativeProxy;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.uimanager.events.RCTModernEventEmitter;

public class NoopEventHandler implements RCTModernEventEmitter {
  @Override
  public void receiveEvent(
      int surfaceId,
      int targetTag,
      @NonNull String eventName,
      boolean canCoalesceEvent,
      int customCoalesceKey,
      @Nullable WritableMap params,
      int category) {
    // noop
  }

  @Override
  public void receiveEvent(
      int surfaceId, int targetTag, @NonNull String eventName, @Nullable WritableMap params) {
    // noop
  }

  @Override
  public void receiveEvent(int targetTag, @NonNull String eventName, @Nullable WritableMap params) {
    // noop
  }

  @Override
  public void receiveTouches(
      @NonNull String eventName,
      @NonNull WritableArray touches,
      @NonNull WritableArray changedIndices) {
    // noop
  }
}

File: packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/nativeProxy/NoopEventHandler.kt
package com.swmansion.reanimated.nativeProxy

import com.facebook.react.bridge.WritableArray
import com.facebook.react.bridge.WritableMap
import com.facebook.react.uimanager.events.RCTModernEventEmitter

class NoopEventHandler : RCTModernEventEmitter {
    override fun receiveEvent(
        surfaceId: Int,
        targetTag: Int,
        eventName: String,
        canCoalesceEvent: Boolean,
        customCoalesceKey: Int,
        params: WritableMap?,
        category: Int,
    ) {
        // noop
    }

    override fun receiveEvent(
        surfaceId: Int,
        targetTag: Int,
        eventName: String,
        params: WritableMap?,
    ) {
        // noop
    }

    override fun receiveEvent(targetTag: Int, eventName: String, params: WritableMap?) {
        // noop
    }

    override fun receiveTouches(
        eventName: String,
        touches: WritableArray,
        changedIndices: WritableArray,
    ) {
        // noop
    }
}
